src/MongoDB.Analyzer/Core/Builders/BuildersResolveVariables.cs

```diff
@@ -0,0 +1,445 @@
+            contextInformation.VariableValues.Add(variableName, variableValue);
+
+            var originalExpression = variableValue.ExpressionAnalysisContext.Node.OriginalExpression;
+
+            if (contextInformation.BuilderNodesProcessed.Add(originalExpression))
+            {
+                contextInformation.VariableValuesList.Add(variableValue);
+            }
+        }
+        else
+        {
+            RemoveVariablesFromContext(new List<string> { variableName }, contextInformation);
+        }
+
+        return evaluationResult.CanEvaluate;
+    }
+
+    private static bool ProcessAssignment(AssignmentExpressionSyntax assignmentExpression, ContextInformation contextInformation, HashSet<string> dirtyVariables, bool displayDiagnostics)
+    {
+        var evaluationResult = PreEvaluate(assignmentExpression, contextInformation, null, displayDiagnostics);
+        var variableName = (assignmentExpression.Left.TrimParenthesis() as IdentifierNameSyntax).Identifier.ValueText;
+
+        if (evaluationResult.CanEvaluate)
+        {
+            var variableValue = evaluationResult.SyntaxNodeToVariableValueMap.Values.First();
+            var generateNewContext = IsCompoundExpression(assignmentExpression.Kind()) || evaluationResult.SyntaxNodeToVariableValueMap.Count() > 1;
+
+            if (generateNewContext)
+            {
+                var expressionAnalysisContext = GenerateNewExpressionAnalysisContext(assignmentExpression, evaluationResult.SyntaxNodeToVariableValueMap);
+                variableValue = new VariableValue(expressionAnalysisContext, new List<Location>());
+            }
+
+            dirtyVariables.Add(variableName);
+            contextInformation.VariableValues[variableName] = variableValue;
+            var originalExpression = variableValue.ExpressionAnalysisContext.Node.OriginalExpression;
+
+            if (contextInformation.BuilderNodesProcessed.Add(originalExpression))
+            {
+                contextInformation.VariableValuesList.Add(variableValue);
+            }
+
+            if (assignmentExpression.Parent is ConditionalExpressionSyntax)
+            {
+                RemoveVariablesFromContext(new List<string>() { variableName }, contextInformation);
+            }
+        }
+        else
+        {
+            RemoveVariablesFromContext(new List<string>() { variableName }, contextInformation);
+        }
+
+        return evaluationResult.CanEvaluate;
+    }
+
+    private static void RemoveVariablesFromContext(IEnumerable<string> variablesToRemove, ContextInformation contextInformation)
+    {
+        var variableValues = contextInformation.VariableValues;
+        var variableValuesList = contextInformation.VariableValuesList;
+        var builderNodesProcessed = contextInformation.BuilderNodesProcessed;
+
+        foreach (var variable in variablesToRemove)
+        {
+            var isValidVariable = variableValues.TryGetValueSafe(variable, out var previousValue);
```

Comment:
You can have just
```
if (variableValues.Remove(variable, out previosValue))
{
   do stuff.
}
```

Reply:
Using "variableValues.Remove(variableName, out var previousValue)" gives me the error "No overload for method 'Remove' takes 2 arguments".  I think it may be that older versions of .NET/ .NET Core don't have this method.